src/cljs/nnts2/events.cljs

```diff
@@ -1,27 +1,41 @@
 (ns nnts2.events
   (:require
-    [re-frame.core :as re-frame]
-    [nnts2.db :as db]
-    [nnts2.user.api :as user-api]))
+   [re-frame.core :as re-frame]
+   [day8.re-frame.http-fx]
+   [nnts2.db :as db]
+   [nnts2.organization.api :as org-api]
+   [nnts2.user.api :as user-api]))
 
 (re-frame/reg-event-db
-  ::initialize-db
-  (fn [_ _]
-    db/default-db))
+ ::initialize-db
+ (fn [_ _]
+   db/default-db))
 
 (re-frame/reg-event-db
-  ::set-active-panel
-  (fn [db [_ active-panel]]
-    (assoc db :active-panel active-panel)))
+ ::set-active-panel
+ (fn [db [_ active-panel]]
+   (assoc db :active-panel active-panel)))
 
 
 (re-frame/reg-event-fx
-  ::get-user-info
-  (fn [_ event]
-    (user-api/get-info)))
+ ::get-user-info
+ (fn [{db :db} event]
+   {:http-xhrio user-api/user-info-map
+    :db (assoc db :loading true)}))
 
+(re-frame/reg-event-fx
+ :create-organization
+ (fn [db [_ org-details]]
+   {:http-xhrio (org-api/create-map org-details)
+    :db (assoc-in db [:create-organization :state] :loading)}))
+
+(re-frame/reg-event-db
+ :organization-created
+ (fn [db [_ org-details]]
+   (prn org-details)
+   (assoc db :organization-list org-details)))
```

Comment:
Adding received org details directly to the list supercedes any order you might want your org list to be in. What we had decided was to do get-org-list again after the creation is successful and then populate the org list from the returned response.

Reply:
In the show-organizations branch (where this is more relevant), I maintain an organization list in the db in a way that order is preserved -- I don't have to do a get-all for orgs, I just update the existing vector of orgs instead. Since the "show" part isn't relevant to this PR, I think it's best for those changes to reside on that branch.